File: PlantHere/PlantHere.Application/CQRS/BasketItem/Commands/CreateBasketItem/CreateBasketItemCommand.cs
using PlantHere.Application.CQRS.Base;

namespace PlantHere.Application.CQRS.BasketItem.Commands.CreateBasketItem
{
    public class CreateBasketItemCommand : CommandBase<CreateBasketItemCommandResult>
    {
        public string ProductId { get; set; }

        public string UserId { get; set; }

        public string ProductName { get; set; }

        public decimal Price { get; set; }

        public decimal DiscountedPrice { get; set; }

        public CreateBasketItemCommand(string productId, string userId, string productName, decimal price, decimal discountedPrice)
        {
            ProductId = productId;
            UserId = userId;
            ProductName = productName;
            Price = price;
            DiscountedPrice = discountedPrice;
        }
    }
}

File: PlantHereNodeJs/Core/PlantHere.Aplication/RequestResponseModels/BasketItem/Commands/CreateBasketItem/CreateBasketItemCommand.js
const Guid = require("../../../../../PlantHere.Aplication/Services/GuidService")

class CreateBasketItemCommand{
    constructor(productId,userId,productName,price,discountedPrice) {
        this.ProductId = productId,
        this.UserId = userId,
        this.ProductName = productName,
        this.Price = price,
        this.DiscountedPrice = discountedPrice,
        this.UniqueId  = Guid.createGuid(),
        this.CreatedDate =  Guid.dateNow(); 
        this.UpdatedDate = Guid.dateNow(); 
    }
}

module.exports =  {CreateBasketItemCommand}